synapse/http/client.py

```diff
@@ -301,9 +299,8 @@ def __init__(
                 we may not request.
             ip_whitelist: The whitelisted IP addresses, that we can
                request if it were otherwise caught in a blacklist.
-            http_proxy: proxy server to use for http connections. host[:port]
-            https_proxy: proxy server to use for https connections. host[:port]
-            no_proxy: locations that should explicitly not use a proxy.
+            use_proxy: Whether proxy settings should be discovered and used
+                from conventional environment variables. Defaults to false.
```

Comment:
```suggestion
                from conventional environment variables.
```

Shouldn't need to specify this; it's listed in the arguments already which IDEs *should* show.